src/Drivers/Zarinpal/Zarinpal.php

```diff
@@ -54,11 +54,21 @@ public function purchase()
             $description = $this->settings->description;
         }
 
+        if (!empty($this->invoice->getDetails()['mobile'])) {
+            $mobile = $this->invoice->getDetails()['mobile'];
+        }
+
+        if (!empty($this->invoice->getDetails()['email'])) {
+            $email = $this->invoice->getDetails()['email'];
```

Comment:
i thinks it would be better to write it like the below:

```php
$email = $this->invoice->getDetails()['email'] ?: ''
```
what's do you think about my idea?

Reply:
I've tested the code and works fine and you could see the Mobile and Email on the Zarinpal's Dashboard. We use that to identify the unverified payments to match them with the user.

> The only thing I should mention here is, in their documentation they say that Mobile and Email are optional but if you only pass one of them it doesn't appear on the dashboard and you've to send both (Email and Mobile) to get them both on the dashboard. This is a bug on their side and I've submitted a ticket with them.

Unfortunately, I won't be able to write some unit test at the moment as I'm trying to move to this package from the original Zarinpal package (As it doesn't maintain anymore). But definitely will contribute to this package more in the future.

I agree with your suggestion and I'll update the code to reflect the suggestion.

Reply:
thanks for your complete descriptions. lets merge it :)